modules/targets/darwin/default.nix

```
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+  inherit (lib.strings) escapeShellArg;
+
+  cfg = config.targets.darwin;
+
+  toDefaultsFile = domain: attrs:
+    pkgs.writeText "${domain}.plist"
+    (lib.generators.toPlist { } (filterAttrs (n: v: v != null) attrs));
+
+  toActivationCmd = domain: attrs:
+    "$DRY_RUN_CMD defaults import ${escapeShellArg domain} ${
+      toDefaultsFile domain attrs
+    }";
+
+  activationCmds = mapAttrsToList toActivationCmd cfg.defaults;
+in {
+  imports = [ ./keybindings.nix ./linkapps.nix ./search.nix ];
+
+  options.targets.darwin = {
+    defaults = mkOption {
+      type = types.submoduleWith { modules = cfg.defaultsSchema; };
+      default = { };
+      example = {
+        "com.apple.desktopservices" = {
+          DSDontWriteNetworkStores = true;
+          DSDontWriteUSBStores = true;
+        };
+      };
+      description = ''
+        Set macOS user defaults. Values set to <literal>null</literal> are
+        ignored.
+
+        <warning>
+          <para>
+            Some settings might require a re-login to take effect.
+          </para>
+        </warning>
+      '';
+    };
+
+    defaultsSchema = mkOption {
+      type = with types; listOf anything;
+      default = [ ];
+      example = [
+        ({ ... }: { option."org.nixos.foo".enable = mkEnableOption "foo"; })
+      ];
```

```suggestion
      example = literalExample ''
        [
          ({ ... }: { option."org.nixos.foo".enable = mkEnableOption "foo"; })
        ]
      '';
```

Probably unnecessary with the comment above.

---

I've now removed this option.